src/classes/class-wordpress.php

```diff
@@ -95,6 +96,20 @@ public function time_now( string $timezone ) : object {
 			return new DateTime( 'now', new DateTimeZone( $timezone ) );
 		}
 
+		/**
+		 * Makes the filesystem read-only.
+		 *
+		 * @param bool   $file_mod_allowed Whether file modifications are allowed.
+		 * @param string $context The usage context.
+		 */
+		public function read_only_filesystem( $file_mod_allowed, $context ) {
+			if ( ! get_option( 'woocart_readonly_filesystem', false ) ) {
+				return true;
+			}
+
+			return false;
```

Comment:
```suggestion
			return get_option( 'woocart_readonly_filesystem', false );
```

Reply:
Updated.

`return ! get_option( 'woocart_readonly_filesystem', false );`

It works the opposite. If the option is set, we should return `false` to lock the filesystem.